client/js/widgets/widget.js

```diff
@@ -599,6 +601,29 @@ export class Widget extends StateManaged {
           }
         }
       }
+      if(a.func == 'IF') {
+        if(typeof a.variable != 'undefined' && Array.isArray(a.thenRoutine)) {
+          const inheritVariables = variables;
```

Comment:
The whole inheritance stuff seems too complicated. I'll play around with it when I'm at a PC but if it works like this, we can also merge it first.